app/src/main/java/com/samco/trackandgraph/antlr/evaluation/DataValuesBase.kt

```diff
@@ -0,0 +1,151 @@
+package com.samco.trackandgraph.antlr.evaluation
+
+import com.samco.trackandgraph.antlr.generated.TnG2Parser
+import com.samco.trackandgraph.database.entity.DataPoint
+import org.antlr.v4.runtime.tree.ParseTree
+import org.threeten.bp.Duration
+
+abstract class Value {
```

Comment:
So we have to parse everything in the data-transformation code into something we can work with (call functions on etc). Therefore there are these Value classes which support + - / * and are the default return values for functions etc. 
Note that Value itself is abstract